### PR TITLE
feat: add history api

### LINKED ISSUE #11: Proposal: History API for listing changes
Partial dependent on the resolution of the discussion in decentralized-identity/confidential-storage#207, below is the proposal for an endpoint for requesting a history of changes from an EDV.

**Proposal**
An HTTP GET endpoint `/history` that supports the following query parameters

- afterSequence: [Unsigned Integer Value] Fetch only changes after the supplied sequence value
- distinctDocs: [Boolean] Fetch the changes filtered to the distinct documents effected, e.g if multiple updates have been applied to one document only a single entry in the return results will be present for that document.

Example request

```
GET /data-vaults/z4sRgBJJLnYy/history?afterSequence=12&distinctDocs=true 
HTTP/1.1
Host: example.com
Content-Type: application/json
Accept: application/json, text/plain, */*
Accept-Encoding: gzip, deflate

[
    {
        "documentId": "https://example/data-vaults/z4sRgBJJLnYy/docs/z4sAsDFrGYh",
        "sequence": 15 // E.g 3 changes have been made to this document since sequence "12" but only "15" is showed due to the distinctDocs parameter
    },
    {
        "documentId": "https://example/data-vaults/z4sRgBJJLnYy/docs/zaCfHeKsNesS",
        "sequence": 16,
        "isDeleted": true
    }
]
```

### PR DESCRIPTION
**Pull request by tplooker**
Resolves #11

### CODE REVIEW

**Review by dlongley — APPROVED** [review]

File: index.html [hunk 1]
```diff
@@ -2285,6 +2285,139 @@ <h3>
...
+  [
+    {
+        "documentId": "https://example/data-vaults/z4sRgBJJLnYy/docs/z4sAsDFrGYh",
+        "sequence": 1
+    },
+    {
+        "documentId": "https://example/data-vaults/z4sRgBJJLnYy/docs/zaCfHeKsNesS",
+        "sequence": 2
+    }
+  ]
```
Comment (dlongley) [inline index.html:2355]: We may want to mention a need for pagination here (just like what we need w/ the find/query API).

File: index.html [hunk 2]
```diff
@@ -2285,6 +2285,139 @@ <h3>
...
+        </table>
+
+          <p>
+  An example request is shown below:
+          </p>
+          
+          <pre class="example highlight" title="Example history API request">
+  GET https://example.com/encrypted-data-vaults/z4sRgBJJLnYy/history HTTP/1.1
+  Host: example.com
+  Content-Range: 0-1048576
```
Comment (tplooker) [inline index.html:2332]: Remove this request element

Comment (tplooker) [inline index.html:2332]:
```suggestion
```

File: index.html (hunk 1 above)
Comment (tplooker) [reply index.html:2354]: Agree @dlongley, I was thinking we leave this until we tackle pagination as a general problem across all APIs in EDVs.

**Review by msporny — APPROVED** [review]

**Pull request merged**